src/object/cli_obj.c

```diff
@@ -1903,9 +1921,17 @@ obj_comp_cb(tse_task_t *task, void *data)
 	if (obj_retry_error(task->dt_result)) {
 		obj_auxi->io_retry = 1;
 		if (task->dt_result == -DER_INPROGRESS)
-			obj_auxi->retry_with_leader = 1;
+			obj_auxi->to_leader = 1;
 	}
 
+	/* If the RPC sponsor set the @spec_shard, then means it wants
+	 * to fetch data from the specified shard. If such shard isn't
+	 * ready for read, we should let the caller know that, but not
+	 * re-direct the fetch RPC to other replica.
+	 */
```

Comment:
if the request just be timedout due to some temporarily network issue, still cannot retry it? maybe better to differentiate the normal network issue and "not ready for read" case?

Reply:
You are right, I will fix it.